bridge/src/config.rs

```diff
@@ -157,7 +159,14 @@ impl Node {
 			concurrent_http_requests,
 		};
 
-		Ok(result)
+		// Ensure that the contract address is specified for non-deploy builds:
+		if cfg!(not(feature = "deploy")) && node.contract_address.is_none() {
+			return Err("Contract address not specified. Please define the 'contract_address' \
```

Comment:
Does it make sense to raise ConfigError here the same as it is done in https://github.com/c0gent/poa-bridge/blob/45d643cb48adcd9d6beec3b9192017d791f18e99/bridge/src/config.rs#L130

Reply:
I'm not exactly sure I understand you. Are asking whether this error check is redundant or unnecessary?

If you're implying that the string error is a bad idea I agree. I'd be happy to simply use the `ConfigError` variant or create a new variant. As I mentioned, I badly want to completely revamp the error handling and remove all string errors and replace them with dedicated variants but I didn't want to include all of that in this pull request.

Let me know what you want me to do with that error for now.

Reply:
Agreed, I'll change it now.

Reply:
thanks